src/aui-button/js/aui-button-core.js

```diff
@@ -65,10 +70,10 @@ ButtonExt.ATTRS = {
      * @writeOnce
      */
     domType: {
-        writeOnce: true,
         validator: function(val) {
-            return val.toLowerCase() === 'button' || val.toLowerCase() === 'submit';
-        }
+            return this._domTypeValidator(val);
```

Comment:
You don't need create a function to the validator just to call a `_domTypeValidator`, instead of you can assign `_domTypeValidator` directly to the validator like this `validator: '_domTypeValidator'.